### PR TITLE
Documentation improvement

### LINKED ISSUE #70: DG - Incorrect links
Currently, most of the links in the developer guide are from the `addressbook`. We will need to check and update it to `SoConnect` links.

Example:
image

`Logic.java` has the wrong link - it is the java file from the `addressbook` 

Not a high priority, just update this before the final submission!

### LINKED ISSUE #124: Week 10: Update DG - Search feature

### PR DESCRIPTION
**Pull request by florentianayuwono**
- Update `search` feature related documentation.
- Improve overall project documentation wording, fix bugs and incorrect links/naming.

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -248,7 +248,7 @@ Example:
 
 ### Clearing all entries : `clear`
 
-Clears all entries from the address book.
+Clears all entries from the SoConnect.
```
Comment (Eugene-Ong-W-X) [inline docs/UserGuide.md]:
```suggestion
Clears all contacts from the SoConnect.
```

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -9,7 +9,9 @@ title: Developer Guide
 
 ## **Acknowledgements**
 
-* {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+* This product is adapted from [SoConnect Level-3](https://nus-cs2103-ay2223s1.github.io/tp/).
```
Comment (Eugene-Ong-W-X) [inline docs/DeveloperGuide.md]:
```suggestion
* This product is adapted from [AddressBook Level-3](https://nus-cs2103-ay2223s1.github.io/tp/).
```

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -23,7 +25,7 @@ Refer to the guide [_Setting up and getting started_](SettingUp.md).
 
 <div markdown="span" class="alert alert-primary">
 
-:bulb: **Tip:** The `.puml` files used to create diagrams in this document can be found in the [diagrams](https://github.com/se-edu/addressbook-level3/tree/master/docs/diagrams/) folder. Refer to the [_PlantUML Tutorial_ at se-edu/guides](https://se-education.org/guides/tutorials/plantUml.html) to learn how to create and edit diagrams.
+:bulb: **Tip:** The `.puml` files used to create diagrams in this document can be found in the [diagrams](https://github.com/se-edu/SoConnect-level3/tree/master/docs/diagrams/) folder. Refer to the [_PlantUML Tutorial_ at se-edu/guides](https://se-education.org/guides/tutorials/plantUml.html) to learn how to create and edit diagrams.
```
Comment (Eugene-Ong-W-X) [inline docs/DeveloperGuide.md]:
```suggestion
:bulb: **Tip:** The `.puml` files used to create diagrams in this document can be found in the [diagrams](https://github.com/AY2223S1-CS2103T-W15-1/tp/tree/master/docs/diagrams/) folder. Refer to the [_PlantUML Tutorial_ at se-edu/guides](https://se-education.org/guides/tutorials/plantUml.html) to learn how to create and edit diagrams.
```

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -243,7 +254,7 @@ The following sequence diagram shows how the search operation works:
 
 </div>
 
-Step 3. The user is shown the searched list. The searched list contains all contacts that are relevant to the user's search query.
+Step 4. The user is shown the searched list. The searched list contains all contacts that are relevant to the user's search query.
```
Comment (Eugene-Ong-W-X) [inline docs/DeveloperGuide.md]: Just a suggestion, maybe `filtered list` might be better instead of `searched list`?

**Review by Eugene-Ong-W-X — CHANGES_REQUESTED** [review]
Nice work! Fixed a broken link for you and some possible typos / better word choice for you to consider.

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (florentianayuwono) [reply docs/DeveloperGuide.md]: Agree!

**Pull request merged**